### PR TITLE
Show progress bar when exporting instances

### LINKED ISSUE #245: Add progress dialog when exporting an instance
### Role

Exporting instances for use with Prism Launcher on other systems

### Suggestion

When exporting an instance, Prism Launcher should display a progress dialog. Currently the entire launcher stops responding while it compresses the instance in the background

### Benefit

This would help a lot when exporting larger instances, especially since data compression is CPU-intensive in nature and may take a while to complete

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

_No response_

### PR DESCRIPTION
**Pull request by Trial97**
fixes #245

### CODE REVIEW

File: launcher/MMCZip.h [hunk 1]
```diff
@@ -35,110 +36,148 @@
...
+    {
+        if (m_cleanUp)
+            delete m_output;
+    }
+
+   protected:
+    virtual void executeTask() override;
+
+   private:
+    QuaZip* m_output;
```
Comment (TheKodeToad) [inline launcher/MMCZip.h]: ...why not use no pointer or smart pointer?

Comment (Trial97) [reply launcher/MMCZip.h]: I want to use the same task for modrinth and curse export so I need a reference for them here, but yeah a share pointer would make more sense

File: launcher/MMCZip.cpp [hunk 2]
```diff
@@ -2,6 +2,7 @@
 /*
  *  PolyMC - Minecraft Launcher
```
Comment (TayouVR) [inline launcher/MMCZip.cpp]:
```suggestion
 *  Prism Launcher - Minecraft Launcher
```

File: launcher/MMCZip.h [hunk 3]
```diff
@@ -2,6 +2,7 @@
 /*
  *  PolyMC - Minecraft Launcher
```
Comment (TayouVR) [inline launcher/MMCZip.h]:
```suggestion
 *  Prism Launcher - Minecraft Launcher
```

**Review by TayouVR — APPROVED** [review]

**Comment by TheKodeToad** [comment]
Hmm, I think this would be better with a future watcher?

**Comment by Trial97** [comment]
Why?
This way is simpler.

**Comment by TheKodeToad** [comment]
I can't remember entirely but I think I had some issues with thread safety... so i used `invokeMethod`  and flowln suggested i used a future afterwards... I think it's nicer

Although it's likely there weren't any thread safety issues and I was just being over-cautious :woman_shrugging:

**Comment by TheKodeToad** [comment]
Ah,
https://github.com/PrismLauncher/PrismLauncher/blob/develop/launcher/tasks/Task.cpp#L157

changing e.g. this on another thread probably would cause problems :|

**Comment by Trial97** [comment]
Ok let me look into it

**Comment by Trial97** [comment]
Yep fixed that

**Review by getchoo — APPROVED** [review]
lgtm! worked nicely on my end as well :)

**Review by Scrumplex — APPROVED** [review]

**Pull request merged**